homework-3/cycles/script.js

```diff
@@ -0,0 +1,71 @@
+// 1) На основе строки “i am in the easycode” сделать новую строку где первые буквы каждого слова будут в верхнем регистре. Использовать for или while.
+
+var string = "i am in the easycode";
+var space = "";
+
+for (var i = 0; i < string.length; i++) {
+  space += (string[i - 1] == ' ') ? string[i].toUpperCase() : string[i];
+}
+
+console.log(space);
+
+// 2) Дана строка “tseb eht ma i”. Используя циклы, сделать строку-перевертыш (то есть последняя буква становится первой, предпоследняя - второй итд).
+
+var a = "tseb eht ma i";
+var b = "";
+
+for (var i = a.length - 1; i >=0; i--) {
+  b += a[i];
+}
+
+console.log(b);
+
+// 3) Факториал числа - произведение всех натуральных чисел от 1 до n
+// включительно: 3! = 3*2*1, 5! = 5*4*3*2*1. С помощью циклов вычислить факториал числа 10. Использовать for.
+
+var n = 10;
+var result = 1;
+
+for (var i = 1; i <=n; i++) {
+  result *= i;
+}
+
+console.log(result);
+
+// 4) На основе строки “JavaScript is a pretty good language” сделать новую строку,
+// где каждое слово начинается с большой буквы, а пробелы удалены. Использовать for.
+
+var c = "JavaScript is a good language";
+var d = "";
+
+for (i = 0; i < c.length; i++) {
+    d += " " != c[i] ? i && " " != c[i - 1] ? c[i] : c[i].toUpperCase() : "";
```

Comment:
Please format like we discussed in lesson;

this variant would be easier
`var c = "JavaScript is a good language";`
`var d = "";`
`for (i = 0 ; i<a.length; i++) {`
	`b += (a[i-1] === ' ') ? (a[i].toUpperCase()) : a[i];`
	`b = b.split(' ').join('');`
`}`